src/Agent.Plugins/PipelineCache/SavePipelineCacheV0.cs

```diff
@@ -11,12 +11,7 @@ public class SavePipelineCacheV0 : PipelineCacheTaskPluginBase
     {
         public override string Stage => "post";
 
-        protected override async Task ProcessCommandInternalAsync(
-            AgentTaskPluginExecutionContext context,
-            Fingerprint fingerprint,
-            Func<Fingerprint[]> restoreKeysGenerator,
-            string path,
-            CancellationToken token)
+        public override async Task RunAsync(AgentTaskPluginExecutionContext context, CancellationToken token)
         {
```

Comment:
I don't get why we need to go from ProcessCommandInternalAsync to overriding RunAsync here

Reply:
So, the RunAsync function runs before the ProcessCommandInternalAsync because of the framework, and the fingerprint logic is currently in there. To resolve the issue - https://github.com/microsoft/azure-pipelines-tasks/issues/10907 , we are checking the condition first, and then creating the fingerprint. I'll add comments to the code.

Reply:
#2374